src/compass/utils/h5_helpers.py

```diff
@@ -0,0 +1,448 @@
+from datetime import datetime
+
+import isce3
+import numpy as np
+from osgeo import osr
+
+import compass
+from compass.utils.raster_polygon import get_boundary_polygon
+
+def init_geocoded_dataset(geocoded_group, dataset_name, geo_grid, dtype,
+                          description):
+    '''
+    Create and allocate dataset for isce.geocode.geocode_slc to write to that
+    is NC compliant
+
+    geocoded_group: h5py.Group
+        h5py group where geocoded dataset will be created in
+    dataset_name: str
+        Name of dataset to be created
+    geo_grid: isce3.product.GeoGridParameters
+        Geogrid of output
+    dtype: str
+        Data type of dataset to be geocoded
+    description: str
+        Description of dataset to be geocoded
+    '''
+    shape = (geo_grid.length, geo_grid.width)
+    geocoded_ds = geocoded_group.require_dataset(dataset_name, dtype=dtype,
+                                                 shape=shape)
+
+    geocoded_ds.attrs['description'] = description
+
+    # Compute x scale
+    dx = geo_grid.spacing_x
+    x0 = geo_grid.start_x + 0.5 * dx
+    xf = x0 + (geo_grid.width - 1) * dx
+    x_vect = np.linspace(x0, xf, geo_grid.width, dtype=np.float64)
+
+    # Compute y scale
+    dy = geo_grid.spacing_y
+    y0 = geo_grid.start_y + 0.5 * dy
+    yf = y0 + (geo_grid.length - 1) * dy
+    y_vect = np.linspace(y0, yf - dy, geo_grid.length, dtype=np.float64)
+
+    # following copied and pasted (and slightly modified) from:
+    # https://github-fn.jpl.nasa.gov/isce-3/isce/wiki/CF-Conventions-and-Map-Projections
+    x_ds = geocoded_group.require_dataset('x', dtype='float64', data=x_vect,
+                                          shape=x_vect.shape)
+    y_ds = geocoded_group.require_dataset('y', dtype='float64', data=y_vect,
+                                          shape=y_vect.shape)
+
+    # Mapping of dimension scales to datasets is not done automatically in HDF5
+    # We should label appropriate arrays as scales and attach them to datasets
+    # explicitly as show below.
+    x_ds.make_scale()
+    geocoded_ds.dims[0].attach_scale(x_ds)
+    y_ds.make_scale()
+    geocoded_ds.dims[1].attach_scale(y_ds)
```

Comment:
```suggestion
    x_ds.make_scale()
    geocoded_ds.dims[1].attach_scale(x_ds)
    y_ds.make_scale()
    geocoded_ds.dims[1].attach_scale(y_ds)
```

(adding it here so it's easier to click `accept suggestion`)